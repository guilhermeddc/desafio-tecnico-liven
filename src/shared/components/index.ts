export * from './UI/LinearDeterminate';
export * from './UI/MenuSelected';
export * from './UI/ProductCard';
export * from './UI/ListItemMenu';
