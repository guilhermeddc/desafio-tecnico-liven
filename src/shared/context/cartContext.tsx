import React, {createContext, useCallback, useMemo, useState} from 'react';

interface IAddToCart {
  product: {
    id: number;
    title: string;
    image: string;
  };
  quantity: number;
  value: number;
}

export interface IContextCart {
  products: IAddToCart[];
  setProducts(value: IAddToCart[]): void;
  cartQuantity: number;
  cartTotalPrice: number;
  addProduct(id: number): void;
  removeProduct(id: number): void;
  deleteProduct(id: number): void;
}

export const CartContext = createContext<IContextCart>({} as IContextCart);

export const CartProvider: React.FC = ({children}) => {
  const [products, setProducts] = useState<IAddToCart[]>([]);

  const cartQuantity = useMemo(() => {
    let quantity = 0;

    products.map((product) => {
      quantity += product.quantity;
    });

    return quantity;
  }, [products]);

  const cartTotalPrice = useMemo(() => {
    let value = 0;

    products.map((product) => {
      value += product.value * product.quantity;
    });

    return value;
  }, [products]);

  const handleAddProduct = useCallback(
    (id: number) => {
      setProducts(
        products.map((p) =>
          p.product.id === id ? {...p, quantity: p.quantity + 1} : p,
        ),
      );
    },
    [products],
  );

  const handleRemoveProduct = useCallback(
    (id: number) => {
      setProducts(
        products.map((p) =>
          p.product.id === id ? {...p, quantity: p.quantity - 1} : p,
        ),
      );
    },
    [products],
  );

  const handleDeleteProduct = useCallback(
    (id: number) => {
      setProducts(products.filter((p) => p.product.id !== id));
    },
    [products],
  );

  return (
    <CartContext.Provider
      value={{
        products,
        setProducts,
        cartQuantity,
        cartTotalPrice,
        addProduct: handleAddProduct,
        removeProduct: handleRemoveProduct,
        deleteProduct: handleDeleteProduct,
      }}>
      {children}
    </CartContext.Provider>
  );
};
