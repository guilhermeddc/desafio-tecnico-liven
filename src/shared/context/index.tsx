import React from 'react';

import {CartProvider} from './cartProvider';

export const AppProvider: React.FC = ({children}) => {
  return <CartProvider>{children}</CartProvider>;
};
