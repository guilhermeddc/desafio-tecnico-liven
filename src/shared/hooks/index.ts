export * from './useCart';

export * from './useFetchProductsForCategories';
export * from './useFetchProducts';
export * from './useFetchProduct';
export * from './useFetchCategories';
