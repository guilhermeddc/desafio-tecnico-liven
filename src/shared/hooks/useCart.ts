import {useContext} from 'react';

import {
  CartContext,
  CartProvider,
  IContextCart,
} from 'shared/context/cartContext';

const useCart = (): IContextCart => {
  const context = useContext(CartContext);

  if (!context) {
    throw new Error('useCart must be used within an CartProvider');
  }

  return context;
};

export {CartProvider, useCart};
