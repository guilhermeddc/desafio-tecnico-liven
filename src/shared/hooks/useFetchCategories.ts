import {useQuery} from 'react-query';

import {productService} from 'shared/services/api/productService';

export const useFetchCategories = () => {
  const {data, error, isLoading, isError} = useQuery('categories', () =>
    productService.getAllCategories(),
  );

  return {data, error, isLoading, isError};
};
