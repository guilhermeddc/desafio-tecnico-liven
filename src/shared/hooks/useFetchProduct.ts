import {useQuery} from 'react-query';

import {productService} from 'shared/services/api/productService';

export const useFetchProduct = (id: string | undefined) => {
  const {data, error, isLoading, isError} = useQuery(['product', id], () =>
    productService.getSingleProduct(id),
  );

  return {data, error, isLoading, isError};
};
