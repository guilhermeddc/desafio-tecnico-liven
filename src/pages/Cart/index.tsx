import React from 'react';
import {useNavigate} from 'react-router-dom';

import {AddRounded, RemoveRounded} from '@mui/icons-material';
import {
  Avatar,
  Button,
  ButtonGroup,
  Grid,
  Stack,
  Table,
  TableBody,
  TableCell,
  TableContainer,
  TableHead,
  TableRow,
  Typography,
} from '@mui/material';
import {useCart} from 'shared/hooks';
import {moneyMask} from 'shared/utils/moneyMask';

const Cart: React.FC = () => {
  const navigate = useNavigate();
  const {products, cartTotalPrice, addProduct, removeProduct, deleteProduct} =
    useCart();

  return (
    <Grid container spacing={8}>
      {products.length > 0 ? (
        <>
          <Grid item xs={12}>
            <Typography variant="h4">Your bag</Typography>
          </Grid>

          <Grid item xs={12}>
            <TableContainer sx={{width: {xs: '80vw', md: '100%'}}}>
              <Table sx={{minWidth: 650, overflowX: 'scroll'}}>
                <TableHead>
                  <TableRow>
                    <TableCell>Product</TableCell>
                    <TableCell>Qty</TableCell>
                    <TableCell></TableCell>
                    <TableCell align="right">Price</TableCell>
                  </TableRow>
                </TableHead>
                <TableBody>
                  {products.map((p) => (
                    <TableRow key={p.product.id}>
                      <TableCell>
                        <Stack direction="row" alignItems="center" spacing={3}>
                          <Avatar sx={{borderRadius: 1}}>
                            <img
                              src={p.product.image}
                              alt={p.product.title}
                              width="100%"
                            />
                          </Avatar>
                          <Typography variant="body1">
                            {p.product.title.slice(0, 50)}
                          </Typography>
                        </Stack>
                      </TableCell>

                      <TableCell>
                        <Stack direction="row" alignItems="center" spacing={3}>
                          <ButtonGroup
                            size="small"
                            aria-label="small button group">
                            <Button
                              data-testid={`remove-product-${p.product.id}`}
                              onClick={() => removeProduct(p.product.id)}
                              disabled={p.quantity === 1}>
                              <RemoveRounded />
                            </Button>

                            <Button>{p.quantity}</Button>

                            <Button
                              data-testid={`add-product-${p.product.id}`}
                              onClick={() => addProduct(p.product.id)}>
                              <AddRounded />
                            </Button>
                          </ButtonGroup>
                        </Stack>
                      </TableCell>

                      <TableCell>
                        <Button
                          data-testid={`delete-product-${p.product.id}`}
                          onClick={() => deleteProduct(p.product.id)}>
                          Delete
                        </Button>
                      </TableCell>

                      <TableCell align="right">
                        <Typography variant="body1">
                          {moneyMask(p.value * p.quantity)}
                        </Typography>
                      </TableCell>
                    </TableRow>
                  ))}
                  <TableRow>
                    <TableCell />
                    <TableCell />
                    <TableCell>
                      <Button disabled>Subtotal</Button>
                    </TableCell>
                    <TableCell align="right">
                      <Typography variant="body1">
                        {moneyMask(cartTotalPrice)}
                      </Typography>
                    </TableCell>
                  </TableRow>
                </TableBody>
              </Table>
            </TableContainer>
          </Grid>

          <Grid item xs={12}>
            <Stack alignItems="flex-end" spacing={8}>
              <Typography variant="h6" color="secondary.main">
                Shipping & taxes calculated at Checkout
              </Typography>

              <Button variant="contained" color="secondary" size="large">
                Checkout
              </Button>
            </Stack>
          </Grid>
        </>
      ) : (
        <>
          <Grid item xs={12}>
            <Stack spacing={8}>
              <Typography variant="h2">Your bag is empty</Typography>

              <Button
                color="secondary"
                size="large"
                sx={{maxWidth: 170}}
                onClick={() => navigate('/')}>
                Return to home
              </Button>
            </Stack>
          </Grid>
        </>
      )}
    </Grid>
  );
};

export default Cart;
