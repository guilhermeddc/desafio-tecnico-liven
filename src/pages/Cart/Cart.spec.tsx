import {BrowserRouter} from 'react-router-dom';

import {fireEvent, render, screen} from '@testing-library/react';
import {AppProvider} from 'shared/context';
import {useCart} from 'shared/hooks';
import {productsInCart} from 'tests/mock';

import Cart from '.';

const mockedUsedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUsedNavigate,
  useLocation: jest.fn(),
}));
jest.mock('shared/hooks', () => ({
  useCart: jest.fn(),
}));

const mockedUseCart = useCart as jest.Mock;

describe('Cart', () => {
  it('should render products length is greater than zero', () => {
    mockedUseCart.mockImplementation(() => ({
      cartTotalPrice: 400,
      products: productsInCart,
      cartQuantity: 3,
      addProduct: jest.fn(),
      removeProduct: jest.fn(),
      deleteProduct: jest.fn(),
    }));

    const {container} = render(
      <BrowserRouter>
        <AppProvider>
          <Cart />
        </AppProvider>
      </BrowserRouter>,
    );

    expect(screen.getByText(/Your bag/i)).toBeInTheDocument();
    expect(screen.getByText(/Product 1/i)).toBeInTheDocument();
    expect(screen.getByText(/Product 2/i)).toBeInTheDocument();
    expect(screen.getByText(/Qty/i)).toBeInTheDocument();
    expect(screen.getByText(/Price/i)).toBeInTheDocument();
    expect(container).toBeInTheDocument();
    expect(container).toMatchSnapshot();

    fireEvent.click(screen.getByTestId('add-product-1'));
    expect(screen.getAllByText(/2/i)[0]).toBeInTheDocument();
    fireEvent.click(screen.getByTestId('remove-product-1'));
    expect(screen.getAllByText(/1/i)[0]).toBeInTheDocument();
    fireEvent.click(screen.getByTestId('delete-product-2'));
    expect(screen.getByText(/Product 1/i)).toBeInTheDocument();
  });

  it('should render products length is equal to zero', () => {
    mockedUseCart.mockImplementation(() => ({
      cartTotalPrice: 0,
      products: [],
      cartQuantity: 0,
    }));

    const {container} = render(
      <BrowserRouter>
        <AppProvider>
          <Cart />
        </AppProvider>
      </BrowserRouter>,
    );

    expect(screen.getByText(/Your bag/i)).toBeInTheDocument();
    expect(screen.getByText(/Your bag is empty/i)).toBeInTheDocument();
    expect(screen.getByText(/Return to home/i)).toBeInTheDocument();
    fireEvent.click(screen.getByText(/Return to home/i));
    expect(container).toBeInTheDocument();
  });
});
