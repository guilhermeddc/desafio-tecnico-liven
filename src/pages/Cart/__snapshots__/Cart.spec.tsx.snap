// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart should render products length is greater than zero 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-8 css-r1hick-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
      >
        Your bag
      </h4>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <div
        class="MuiTableContainer-root css-rorn0c-MuiTableContainer-root"
      >
        <table
          class="MuiTable-root css-quj9j6-MuiTable-root"
        >
          <thead
            class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                scope="col"
              >
                Product
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                scope="col"
              >
                Qty
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                scope="col"
              />
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-1azl6jz-MuiTableCell-root"
                scope="col"
              >
                Price
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
          >
            <tr
              class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                <div
                  class="css-a2l1ks-MuiStack-root"
                >
                  <div
                    class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-zcpled-MuiAvatar-root"
                  >
                    <img
                      alt="Product 1"
                      src="https://via.placeholder.com/150"
                      width="100%"
                    />
                  </div>
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                  >
                    Product 1
                  </p>
                </div>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                <div
                  class="css-a2l1ks-MuiStack-root"
                >
                  <div
                    aria-label="small button group"
                    class="MuiButtonGroup-root MuiButtonGroup-outlined css-iajp3t-MuiButtonGroup-root"
                    role="group"
                  >
                    <button
                      class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButtonBase-root MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary css-1k23hlb-MuiButtonBase-root-MuiButton-root"
                      data-testid="remove-product-1"
                      tabindex="0"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="RemoveRoundedIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M18 13H6c-.55 0-1-.45-1-1s.45-1 1-1h12c.55 0 1 .45 1 1s-.45 1-1 1z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                    <button
                      class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButtonBase-root MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary css-1k23hlb-MuiButtonBase-root-MuiButton-root"
                      tabindex="0"
                      type="button"
                    >
                      2
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                    <button
                      class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButtonBase-root MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary css-1k23hlb-MuiButtonBase-root-MuiButton-root"
                      data-testid="add-product-1"
                      tabindex="0"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="AddRoundedIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M18 13h-5v5c0 .55-.45 1-1 1s-1-.45-1-1v-5H6c-.55 0-1-.45-1-1s.45-1 1-1h5V6c0-.55.45-1 1-1s1 .45 1 1v5h5c.55 0 1 .45 1 1s-.45 1-1 1z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                <button
                  class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
                  data-testid="delete-product-1"
                  tabindex="0"
                  type="button"
                >
                  Delete
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  $200.00
                </p>
              </td>
            </tr>
            <tr
              class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                <div
                  class="css-a2l1ks-MuiStack-root"
                >
                  <div
                    class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-zcpled-MuiAvatar-root"
                  >
                    <img
                      alt="Product 2"
                      src="https://via.placeholder.com/150"
                      width="100%"
                    />
                  </div>
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                  >
                    Product 2
                  </p>
                </div>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                <div
                  class="css-a2l1ks-MuiStack-root"
                >
                  <div
                    aria-label="small button group"
                    class="MuiButtonGroup-root MuiButtonGroup-outlined css-iajp3t-MuiButtonGroup-root"
                    role="group"
                  >
                    <button
                      class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButtonBase-root Mui-disabled MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary css-1k23hlb-MuiButtonBase-root-MuiButton-root"
                      data-testid="remove-product-2"
                      disabled=""
                      tabindex="-1"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="RemoveRoundedIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M18 13H6c-.55 0-1-.45-1-1s.45-1 1-1h12c.55 0 1 .45 1 1s-.45 1-1 1z"
                        />
                      </svg>
                    </button>
                    <button
                      class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButtonBase-root MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary css-1k23hlb-MuiButtonBase-root-MuiButton-root"
                      tabindex="0"
                      type="button"
                    >
                      1
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                    <button
                      class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButtonBase-root MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary css-1k23hlb-MuiButtonBase-root-MuiButton-root"
                      data-testid="add-product-2"
                      tabindex="0"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="AddRoundedIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M18 13h-5v5c0 .55-.45 1-1 1s-1-.45-1-1v-5H6c-.55 0-1-.45-1-1s.45-1 1-1h5V6c0-.55.45-1 1-1s1 .45 1 1v5h5c.55 0 1 .45 1 1s-.45 1-1 1z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                <button
                  class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
                  data-testid="delete-product-2"
                  tabindex="0"
                  type="button"
                >
                  Delete
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  $200.00
                </p>
              </td>
            </tr>
            <tr
              class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              />
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              />
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                <button
                  class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root Mui-disabled  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
                  disabled=""
                  tabindex="-1"
                  type="button"
                >
                  Subtotal
                </button>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  $400.00
                </p>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <div
        class="css-gydppj-MuiStack-root"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 css-6zgmsr-MuiTypography-root"
        >
          Shipping & taxes calculated at Checkout
        </h6>
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButtonBase-root  css-1xnmwiw-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Checkout
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
